website/notifications/model.py

```diff
@@ -4,18 +4,35 @@
 from modularodm.exceptions import ValidationValueError
 
 from website.project.model import Node
-from website.notifications.constants import NOTIFICATION_TYPES
+from website.notifications.constants import NOTIFICATION_TYPES, NODE_SUBSCRIPTIONS_AVAILABLE, USER_SUBSCRIPTIONS_AVAILABLE
+import re
 
 
 def validate_subscription_type(value):
     if value not in NOTIFICATION_TYPES:
         raise ValidationValueError
 
 
+def validate_event_type(value):
+    """
+
+    :param value:
+
+    Checks if the value passed in is in the NODE_SUBSCRIPTIONS_AVAILABLE or USER_SUBSCRIPTIONS_AVAILABLE
+    if it is an event name.  Since the only things from NODE_SUBSCRIPTIONS_AVAILABLE that contains "file_updated"
+    is preceded by a user ID, a regex checker was added for that.
+    """
+    prefix = re.compile('[[a-zA-Z0-9]*_]?')
```

Comment:
Matching strings are things like "xyz42_" or anything that has any
combination of letters and numbers followed by and underscore 0 or 1 times.
This is because sometimes "files_updated" is passed, and other times a
users ID, and underscore, and then "files_updated" is passed.

On Wednesday, August 3, 2016, Brian J. Geiger <notifications@github.com>
wrote:

> In website/notifications/model.py
> <https://github.com/CenterForOpenScience/osf.io/pull/6026#discussion_r73417345>
> :
>
> >
> >
> >  def validate_subscription_type(value):
> >      if value not in NOTIFICATION_TYPES:
> >          raise ValidationValueError
> >
> >
> > +def validate_event_type(value):
> > +    """
> > +
> > +    :param value:
> > +
> > +    Checks if the value passed in is in the NODE_SUBSCRIPTIONS_AVAILABLE or USER_SUBSCRIPTIONS_AVAILABLE
> > +    if it is an event name.  Since the only things from NODE_SUBSCRIPTIONS_AVAILABLE that contains "file_updated"
> > +    is preceded by a user ID, a regex checker was added for that.
> > +    """
> > +    prefix = re.compile('[[a-zA-Z0-9]*_]?')
>
> I'm not convinced that this is exactly doing what you want it to, and we'd
> need some tests for this. What are some examples of matching strings?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/CenterForOpenScience/osf.io/pull/6026/files/9f6320d17d9ba489ffddce6590144d5e9c3091e6#r73417345>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AP4VDaAKpM6R5pcTqsA5BFiVhDlyqU0Jks5qcP35gaJpZM4JUReH>
> .
>


--